### PR TITLE
Use TWIT_paginate_cursor in more places

### PR DESCRIPTION
**Pull request by hadley**
Replacing hand-rolled pagination code. This increases consistency and is one step towards #510.

### CODE REVIEW

**Comment by hadley** [comment]
@llrs I'm having a hard time getting collections to work, because their APIs are so different. What do you think about just getting rid of `lookup_collections()` and `get_collections()`? I'd never heard of a twitter collection before this, and my googling doesn't find much about them.

File: R/lists_subscribers.R [hunk 1]
```diff
@@ -42,64 +32,28 @@ lists_subscribers <- function(list_id = NULL,
...
-  params <- lists_params(
-    list_id = list_id,
-    slug = slug,
-    owner_user = owner_user,
-    count = n,
-    cursor = cursor
+  r <- TWIT_paginate_cursor(token, "/1.1/lists/subscribers", params, 
+    n = n,
+    cursor = cursor, 
+    page_size = 200,
```
Comment (llrs) [inline R/lists_subscribers.R:44]: Again, the page_size is less than the maximum allowed (5000) on the documentation. 
```suggestion
    page_size = 5000,
```

File: R/lists_subscriptions.R [hunk 2]
```diff
@@ -31,64 +22,21 @@ lists_subscriptions <- function(user,
...
-  } else {
-    next_cursor <- "0"
-  }
-
-  ## parse to data frame
-  if (parse && has_name_(r, "lists")) {
-    r <- as_lists_subscriptions(r$lists)
-    r <- as.data.frame(r)
+  r <- TWIT_paginate_cursor(token, "/1.1/lists/subscriptions", params,
+    page_size = 20,
```
Comment (llrs) [inline R/lists_subscriptions.R:33]: Page_size limit is 1000 according to the documentation
```suggestion
    page_size = 1000,
```

**Review by llrs — APPROVED** [review]
Thanks for adding more API references I missed.

With this PR I finally I understood how useful TWIT_paginate_cursor is. 

Most comments are to increase the number of results returned on each page to minimize the number of requests.

File: R/http.R [hunk 3]
```diff
@@ -148,6 +148,7 @@ TWIT_paginate_cursor <- function(token, api, params,
                                  n = 5000, 
                                  page_size = 5000, 
                                  cursor = "-1", 
+                                 get_id = function(x) x$ids,
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/http.R:152]: I realised that this should have the same name as the `TWIT_paginate_max_id` function, making it easier to understand. (Those functions are already marked as internal).

File: R/lists_memberships.R [hunk 4]
```diff
@@ -30,104 +30,31 @@ lists_memberships <- function(user = NULL,
...
-  args <- list(
-    cursor = cursor,
-    filter_to_owned_lists = filter_to_owned_lists,
-    token = token,
-    parse = parse,
-    previous_cursor = previous_cursor
+  r <- TWIT_paginate_cursor(token, "/1.1/lists/memberships", params, 
+    n = n,
+    cursor = cursor, 
+    page_size = 200,
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/lists_memberships.R]: Thanks for catching these; I just forgot to update them from the docs.

**Pull request merged**